skimage/measure/_colocalization.py

```diff
@@ -0,0 +1,295 @@
+    """Manders' colocalization coefficient between two channels.
+
+    Parameters
+    ----------
+    image : (M, N) ndarray
+        Image of channel A.
+    mask : (M, N) ndarray of dtype bool
+        Binary mask with segmented regions of interest in channel B.
+        Must have same dimensions as `image`.
+    roi : (M, N) ndarray of dtype bool, optional
+        Only `image0` pixel values within this region of interest mask are
+        included in the calculation.
+        Must have same dimensions as `image`.
+
+    Returns
+    -------
+    mcc : float
+        Manders' colocalization coefficient.
+
+    Notes
+    -------
+    Manders' Colocalization Coefficient (MCC) is the fraction of total
+    intensity of a certain channel that is within the segmented region of a
+    second channel [1]_. It ranges from 0 for no colocalisation to 1 for
+    complete colocalization. It is also referred to as M1 and M2.
+
+    MCC is commonly used to measure the colocalization of a particular protein
+    in a subceullar compartment. Typically a threshold on the second channel is
+    set and values above this are used in the MCC calculation.In this
+    implementation, the thresholded image is provided as the argument
+    `image1_mask`. Alternative segmentation methods on the second channel can
+    also be performed, with the resulting image mask being provided as
+    `image1_mask`.
+
+    The implemented equation is:
+
+    .. math::
+        r = \frac{\sum A_{i,coloc}}{\sum A_i}
+
+    where
+        :math:`A_i` is the value of the :math:`i^{th}` pixel in `image`
+        :math:`A_{i,coloc} = A_i` if math:`Bmask_i > 0`
+        :math:`Bmask_i` is the value of the :math:`i^{th}` pixel in
+        `mask`
+
+    MCC is sensitive to noise, with diffuse signal in the first channel
+    inflating its value. Images should be processed to remove out of focus and
+    background light before the MCC is calculated [2]_.
+
+    References
+    -------
+    .. [1] Manders, E.M.M., Verbeek, F.J. and Aten, J.A. (1993), Measurement of
+           co-localization of objects in dual-colour confocal images. Journal
+           of Microscopy, 169: 375-382.
+           https://doi.org/10.1111/j.1365-2818.1993.tb03313.x
+           https://imagej.net/media/manders.pdf
+    .. [2] Dunn, K. W., Kamocka, M. M., & McDonald, J. H. (2011). A practical
+           guide to evaluating colocalization in biological microscopy.
+           American journal of physiology. Cell physiology, 300(4), C723–C742.
+           https://doi.org/10.1152/ajpcell.00462.2010
+
+    """
+    image = np.asarray(image)
+    mask = is_binary_ndarray(mask)
+    if roi is not None:
+        roi = is_binary_ndarray(roi)
+        check_shape_equality(image, mask, roi)
+        image = image[roi]
+        mask = mask[roi]
+    else:
+        check_shape_equality(image, mask)
+
+    if (np.sum(image) == 0):
+        return 0
+    return np.sum(np.multiply(image, mask)) / np.sum(image)
```

Comment:
```suggestion
    s = np.sum(image)
    if s == 0:
        return 0
    return np.sum(image * mask) / s
```
(to avoid computing `np.sum(image)` twice)

Also tried using `np.sum(image[mask])` in place of `np.sum(image * mask)` but it is substantially slower